src/app/shared/http-services/auth.http-service.ts

```diff
@@ -20,10 +21,31 @@ const longAuthServicesTimeout = 10000;
 export class AuthHttpService {
 
   private http: HttpClient; // an http client specific to this class, skipping all http interceptors
+  private apiUrl = new URL(appConfig.apiUrl);
+  private isSameDomain = getDomain(appConfig.apiUrl) === getDomain(globalThis.location.href);
   private cookieParams = appConfig.authType === 'cookies' ? {
     use_cookie: '1',
-    cookie_secure: appConfig.secure ? '1' : '0',
-    cookie_same_site: appConfig.sameSite ? '1' : '0',
+    cookie_secure: this.apiUrl.protocol === 'https:' || this.apiUrl.hostname === 'localhost' ? '1' : '0',
+    cookie_same_site: '1',
+    /**
+     * NOTE on cookie_same_site:
+     * The 'same-site' policy only applies for auth by third-party cookie.
+     * A cookie party is defined by its domain:
+     * - first-party cookies have domain "dev.algorea.org", "api.algorea.org" or "algorea.org"
+     * - third-party cookies have domain or "complete-other-site.com"
+     * Safari and Firefox both disable third-party cookies by default, and Chrome will in 2021 (but postponed for 2023)
+     * Since third-party cookies will be removed in a near future, we chose not to enable this type of authentication,
+     * making the 'same-site' policy always true.
+     */
+  } : undefined;
+  private requestConfig = appConfig.authType === 'cookies' ? {
+    withCredentials: !this.isSameDomain,
```

Comment:
`withCredentials` looks safe to set in all cookie cases, same domain or not, as it just marks the request as having credentials. (and it might save us issue with future changes in browsers)

Reply:
Fixed.